lndmanage/lib/rebalance.py

```diff
@@ -130,10 +129,6 @@ def _rebalance(
                 #  and blacklist it, to ignore it in the next path.
                 pass
 
-            if effective_fee_rate > self.max_effective_fee_rate:
```

Comment:
Realized not that this introduces another bug; budget can be exceeded if amount is reduced in subsequent trials.

@bitromortac I notice that one thing that makes this harder (for me) to follow and get consistent is that parameters like `budget_sat` and `max_effective_fee_rate` are stored on `self`. It takes tracing through all possible code paths to understand where and in what order it is modified (the classic peril of global mutable state).

IMO one refactor that could make this a lot more clear would be to only keep them in local scope, and pass them as function parameters where needed instead. So a bit more of a functional approach. What do you think?